Sources/NIO/NonBlockingFileIO.swift

```diff
@@ -367,13 +425,21 @@ public struct NonBlockingFileIO {
 
         return self.threadPool.runIfActive(eventLoop: eventLoop) {
             var buf = buffer
+            
             while byteCount > 0 {
                 let n = try buf.readWithUnsafeReadableBytes { ptr in
                     precondition(ptr.count == byteCount)
-                    let res = try fileHandle.withUnsafeFileDescriptor { descriptor in
-                        try Posix.write(descriptor: descriptor,
-                                        pointer: ptr.baseAddress!,
-                                        size: byteCount)
+                    let res: IOResult<ssize_t> = try fileHandle.withUnsafeFileDescriptor { descriptor in
+                        if let toOffset = toOffset {
+                            return try Posix.pwrite(descriptor: descriptor,
+                                                    pointer: ptr.baseAddress!,
+                                                    size: byteCount,
+                                                    offset: off_t(toOffset))
```

Comment:
this isn't correct. If `pwrite` only writes parts of the bytes, we need to increase `offset` for the subsequent calls. So you need to remember how many bytes have been written previously and add that to `toOffset`.

Reply:
Thanks, Weissi, that was certainly problematic. Any suitable way to unit test this?